canopeum_backend/canopeum_backend/models.py

```diff
@@ -156,6 +156,22 @@ class Site(models.Model):
     announcement = models.ForeignKey(Announcement, models.SET_NULL, blank=True, null=True)
     image = models.ForeignKey(Asset, models.SET_NULL, blank=True, null=True)
 
+    def get_plant_count(self) -> int:
+        site_species = Sitetreespecies.objects.filter(site=self)
+        return sum(specie.quantity for specie in site_species)
+
+    def get_sponsor_progress(self) -> float:
+        total_plant_count = self.get_plant_count()
+        if total_plant_count == 0:
+            return 0
+
+        batches = Batch.objects.filter(site=self)
+        sponsored_plant_count = sum(batch.plant_count() for specie in batches)
```

Comment:
My bad, I mis-copy-pasted
```suggestion
        sponsored_plant_count = sum(batch.plant_count() for batch in batches)
```